### DISCUSSION TITLE
Größe der ZRAM Partition identisch mit der Größe des verfügbaren RAM

### DISCUSSION THREAD

**Discussion by r4m3u5** [OpenCCU/OpenCCU#2253] [🇩🇪 Fragen&Antworten] [answered]
### Describe the issue you are experiencing

Die Größe von ZRAM0 entspricht der Größe des verfügbaren Hauptspeichers. In meinem Fall 4 GB auf zwei RPi 4 Model B / 4 GB.

### Describe the behavior you expected

Per Default sollte die Größe von ZRAM0 bei der Hälfte des vorhandenen Hauptspeichers liegen. Soweit ich mich erinnern kann, war das auf dem RPi 2 vor der Migration auf den RPi 4 auch der Fall. Leider kann ich nicht sagen, ob/wann es bei welcher RPi 4 Vorgängerversionen auch so war.

### Steps to reproduce the issue

1. `# free -h`
2. `# zramctl`

### What is the version this bug report is based on?

3.67.10.20230225

### Which base platform are you running?

rpi4 (RaspberryPi4)

### Which HomeMatic/homematicIP radio module are you using?

HM-MOD-RPI-PCB

### Anything in the logs that might be useful for us?

```txt
root@homematic-ccu:~# free -h
              total        used        free      shared  buff/cache   available
Mem:           3.7G      489.9M        3.0G        7.7M      294.6M        3.2G
Swap:          3.9G           0        3.9G
root@homematic-ccu:~# zramctl
NAME       ALGORITHM DISKSIZE DATA COMPR TOTAL STREAMS MOUNTPOINT
/dev/zram0 lz4           3.9G   4K   64B    4K       4 [SWAP]
root@homematic-ccu:~#
```

### Additional information

Im Normalbetrieb sollte das noch kein Problem darstellen, da kein Speicher allokiert wird - zumindest solange SWAP nicht genutzt wird.
Bei 4 GB RAM halte ich das auch für recht unwahrscheinlich, allerdings wäre ZRAM0 == 1/2 Hauptspeicher die "saubere" Variante.

**Comment by jens-maus** [comment, marked as answer]
Bei dem von dir festgestellten Verhalten bzgl. zram als swap übersiehst du leider, das in der tat nur 50% des verfügbaren RAM für das zram-basierte SWAP verwendet wird. Das ganze Setup bzgl. ZRAM als swap kannst du aus folgender Datei herauslesen:

https://github.com/jens-maus/RaspberryMatic/blob/master/buildroot-external/overlay/base-raspmatic/etc/init.d/S01InitZRAMSwap

In in Zeile 10-12 findest du dann genau das was du erwartest:

https://github.com/jens-maus/RaspberryMatic/blob/6e3ef2bf5c1cf8dfaa3534108e3a1d843a9014c7/buildroot-external/overlay/base-raspmatic/etc/init.d/S01InitZRAMSwap#L10-L12

Dieser Startup-Skript basiert auf folgenden Erkenntnissen/Projekt:

https://github.com/StuartIanNaylor/zram-swap-config

Und wenn du dir das ganze noch einmal durch den Kopf gehen lässt, dann erscheint es in der Tat logisch das unter der Ausgae von `free` zwar 4GB erscheint und damit man vermeintlich denken würde/könnte das doch 100% des verfügbaren RAM genutzt wird. Dem ist aber nicht so, eben weil ZRAM ja den `lz4` kompressionsalgorithmus verwendet und man damit eben run 50% mehr Memory bzw. SWAP dadurch ja gewinnt. Und auch wenn du jetzt die vollen 4GB dort im Swap abgelegt sehen würdest, so wären in Echt eben nur 2GB deines RAMs belegt, eben weil die Kompressions von zram ja zum tragen kommt.

Ergo ist das hier auch kein Bug oder Issue, sondern eben ein Feature bzw. by-design so gedacht und deshalb hab ich das auch entsprechend in einen Diskussionsbeitrag konvertiert.

**Comment by r4m3u5** [comment]
Moin Jens,

ja, vollständig überzeugt. Hätte ich mir das Init-Script angeschaut, wäre es klar gewesen.
Danke und keep up the good work!

Cheers
/r
